app/src/main/java/com/adc/gpai/home/ForecasterScreen.kt

```diff
@@ -78,287 +92,729 @@ fun ForecasterScreen() {
+                .width(50.dp)
+        ) {
+            Surface(
+                shape = RoundedCornerShape(6.dp),
+                color = when (course.grade.firstOrNull() ?: '?') {
+                    'A', 'S' -> Color(0xFF4CAF50) // Green
+                    'B' -> Color(0xFF8BC34A) // Light Green
+                    'C' -> Color(0xFFFFC107) // Yellow
+                    'D' -> Color(0xFFFF9800) // Orange
+                    else -> Color(0xFFF44336) // Red for F
+                },
+                modifier = Modifier.menuAnchor()
+            ) {
+                Text(
+                    text = course.grade,
+                    textAlign = TextAlign.Center,
+                    color = Color.White,
+                    fontWeight = FontWeight.Bold,
+                    fontSize = 14.sp,
+                    modifier = Modifier.padding(horizontal = 10.dp, vertical = 6.dp)
+                )
+            }
+            
+            ExposedDropdownMenu(
+                expanded = gradeExpanded,
+                onDismissRequest = { gradeExpanded = false },
+                modifier = Modifier.width(50.dp)
+            ) {
+                gradeOptions.forEach { grade ->
+                    DropdownMenuItem(
+                        text = { 
+                            Text(
+                                text = grade,
+                                modifier = Modifier.fillMaxWidth()
+                            ) 
+                        },
+                        onClick = {
+                            onGradeChange(grade)
+                            gradeExpanded = false
+                        },
+                        modifier = Modifier.fillMaxWidth()
+                    )
+                }
             }
         }
 
-        Spacer(modifier = Modifier.height(24.dp))
-
-        if (openPopup) {
-            DisplayCourseEntryFields(
-                onDismiss = { openPopup = false },
-                onConfirmation = { course: Course ->
-                    viewModel.addCourse(term.id, course.copy(id = term.courses.maxOf { it.id } + 1))
-                    openPopup = false
-                })
+        // Action buttons
+        IconButton(onClick = onEdit) {
+            Icon(
+                imageVector = Icons.Outlined.Edit,
+                contentDescription = "Edit Course",
+                tint = MaterialTheme.colorScheme.primary,
+                modifier = Modifier.size(20.dp)
+            )
         }
 
-        HorizontalDivider()
+        IconButton(onClick = onDelete) {
+            Icon(
+                imageVector = Icons.Outlined.Delete,
+                contentDescription = "Delete Course",
+                tint = MaterialTheme.colorScheme.error,
+                modifier = Modifier.size(20.dp)
+            )
+        }
     }
 }
 
-/*
- Function to display dialog box/popup for new course entry
- */
+// Reusing dialog components from ModifyTranscriptScreen
```

Comment:
CourseDialog and CourseDialogWithTermSelection seem to be copied from ModifyTranscriptScreen with no changes other than whitespace and comments. Why not just import that and use it here instead of copying it?